src/rez/tests/test_test.py

```diff
@@ -0,0 +1,55 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright Contributors to the Rez Project
+
+"""
+test rez package.py unit tests
+"""
+from rez.tests.util import TestBase, TempdirMixin
+from rez.resolved_context import ResolvedContext
+from rez.package_test import PackageTestRunner
+
+class TestTest(TestBase, TempdirMixin):
+    @classmethod
+    def setUpClass(cls):
+        TempdirMixin.setUpClass()
+
+        packages_path = cls.data_path("builds", "packages")
+        cls.settings = dict(
+            packages_path = [packages_path],
+            package_filter=None,
+            implicit_packages=[],
+            warn_untimestamped=False,
+            resolve_caching=False
+        )
+    @classmethod
+    def tearDownClass(cls):
+        TempdirMixin.tearDownClass()
+
+    def test_1(self):
+        """package.py unit tests are correctly run in a testing environment"""
+        context = ResolvedContext(["testing_obj"])
```

Comment:
```suggestion
        self.inject_python_repo()
        context = ResolvedContext(["testing_obj"])
```
This will add a usable `python` package and make it available to the tests (only this test). Without this, the tests won't be portable across platforms because we can't rely on python being available on the system. We need python to run the test suite, but that doesn't mean that calling `python` should work because it doesn't have to be on `PATH`, and by experience, it will sometimes not be on PATH.

The magic happens in https://github.com/AcademySoftwareFoundation/rez/blob/9094b089e4bec6048a12cf0e3d105af5935c8758/src/rez/cli/selftest.py#L89-L93